src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -192,16 +195,11 @@ where
     let id = config.id;
     info!(
         %id,
-        resume_upper = %config.resume_upper.pretty(),
+        as_of = %config.as_of.pretty(),
         "timely-{worker_id} building source pipeline",
     );
 
-    let reclock_follower = {
-        let upper_ts = config.resume_upper.as_option().copied().unwrap();
-        // Same value as our use of `derive_new_compaction_since`.
-        let as_of = Antichain::from_elem(upper_ts.saturating_sub(1));
```

Comment:
Where's this `saturating_sub` happening now?

Reply:
Aha! Elsewhere you're setting the `as_of` to be the `since`, which is at least "stepped back by one", but possibly even further behind, right?